### DISCUSSION TITLE
Moving Away From Kitty (macOS)

### DISCUSSION THREAD

**Discussion by karimlevallois** [wezterm/wezterm#4787] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20230712-072601,f4abf8fd

### Ask your question!

I’m looking to move away from Kitty and I’m trying to replicate my current set up in Wezterm. It’s going pretty well following the online docs, but I have a couple of things that I’m struggling with.

I want to use a key combination that will spawn a new window in maximise (macOS), change the font size to 14 and open Neovim (well LunarVim really) with this new window taking focus. My normal font size is 12, but I prefer larger font size when coding. I don’t want to change the initial launch window style or size.

I’m getting some funky results in the scrollbar when typing commands. Say I start trying lvim, I will get lvim auto suggested then as I continue typing it will not stop at lvim once that’s complete, but will show the command elongated such as lvimmm lvim lviim and looks really messy. Has anyone else seen this before?

**Comment by karimlevallois** [comment]
<img width="572" alt="Screenshot 2024-01-14 at 16 04 28" src="https://github.com/wez/wezterm/assets/80129273/d1ce996b-e72b-43ff-a52e-165c6d9d50ca">

This is the auto-completion issue I am getting.

**Reply by wez** [reply]
No idea, sounds like something weird with the shell line editor. The closest thing I've heard of is this nushell bug:

* https://github.com/wez/wezterm/issues/2779

**Reply by karimlevallois** [reply]
Thanks for coming back on this Wez! I managed to fix the autocomplete issue by using these instructions

https://wezfurlong.org/wezterm/config/lua/config/term.html#:~:text=You%20can%20then%20set%20term,underlines%20(eg%3A%20undercurl).

 It was config.term that was breaking it and causing the duplicated data.

Any idea on my first question?

‘I want to use a key combination that will spawn a new window in maximise (macOS), change the font size to 14 and open Neovim (well LunarVim really) with this new window taking focus. My normal font size is 12, but I prefer larger font size when coding. I don’t want to change the initial launch window style or size.‘

**Reply by cynicaljoy** [reply]
I think you're looking for something like this:

```lua
wezterm.on("open-nvim", function(window, _)
	local _, _, nvim_window = mux.spawn_window({
		args = { "nvim" },
	})
	local overrides = window:get_config_overrides() or {}
	overrides.font_size = 14
	nvim_window:gui_window():set_config_overrides(overrides)
	nvim_window:gui_window():toggle_fullscreen()
end)

config.keys = {
	{
		key = "n",
		mods = "SUPER",
		action = wezterm.action.EmitEvent("open-nvim"),
	},
}

```

**Reply by karimlevallois** [reply]
Thanks for the suggestion, I can get it to work with nvim if I quote the path instead of "nvim" (installed via Homebrew so not in the standard path). 

```
wezterm.on("open-nvim", function(window, _)
	local _, _, nvim_window = mux.spawn_window({
		args = { "/opt/homebrew/bin/nvim" },
	})
	local overrides = window:get_config_overrides() or {}
	overrides.font_size = 14
	nvim_window:gui_window():set_config_overrides(overrides)
	nvim_window:gui_window():toggle_fullscreen()
end)

config.keys = {
	{
		key = "n",
		mods = "SHIFT|CTRL",
		action = wezterm.action.EmitEvent("open-nvim"),
	},
}
```

However I am getting the below issue if trying to use the lvim path?

Unable to spawn zsh /Users/Karim/.local/bin/lvim because:
No viable candidates found in PATH "/usr/bin:/bin:/usr/sbin:/sbin"
⚠️  Process "'zsh /Users/Karim/.local/bin/lvim'" in domain "local" didn't exit cleanly
Exited with code 1.
This message is shown because exit_behavior="CloseOnCleanExit"

When doing the following: -

```
wezterm.on("open-lvim", function(window, _)
	local _, _, nvim_window = mux.spawn_window({
		args = { "zsh /Users/Karim/.local/bin/lvim" },
	})
	local overrides = window:get_config_overrides() or {}
	overrides.font_size = 14
	nvim_window:gui_window():set_config_overrides(overrides)
	nvim_window:gui_window():toggle_fullscreen()
end)

config.keys = {
	{
		key = "l",
		mods = "SHIFT|CTRL",
		action = wezterm.action.EmitEvent("open-lvim"),
	},
}
```

**Reply by cynicaljoy** [reply]
```diff
-		args = { "zsh /Users/Karim/.local/bin/lvim" },
+		args = { "/Users/Karim/.local/bin/lvim" },
```

**Reply by karimlevallois** [reply]
I actually tried without zsh initially and it wouldn't even spawn the window, so that's why I tried zsh with it, like I have to do in Kitty.

I can get it to work with this workaround: -

```
config.keys = {
	{
		key = 'l',
		mods = 'SHIFT|CTRL',
		action = act.Multiple {
			wezterm.action.SpawnWindow,
			wezterm.action.ToggleFullScreen,
			wezterm.action.IncreaseFontSize,
			wezterm.action.IncreaseFontSize,
			wezterm.action.SendString 'lvim',
			wezterm.action.SendKey { key = 'Enter' },
			wezterm.action.ActivateWindowRelative(1),
		},
	},
}
```

However, the font resizing is rough.

**Reply by karimlevallois** [reply]
Is there a way to merge your initial suggestion but with the actions workaround that I have done? Or is there a way to add lvim path to your suggestion?

Appreciate all your help on this.

**Reply by cynicaljoy** [reply]
my bad, try:

```diff
-		args = { "zsh /Users/Karim/.local/bin/lvim" },
+		args = { "zsh -c 'lvim'" },
```

[spawn](https://wezfurlong.org/wezterm/cli/cli/spawn.html#synopsis) calls out:

> Instead of executing your shell, run PROG. For example: `wezterm cli spawn -- bash -l` will spawn bash as if it were a login shell

so you can't just put your command in there because it doesn't have a shell, so you can use `zsh -c 'lvim'`

```
Usage: zsh [<options>] [<argument> ...]

Special options:
  --help     show this message, then exit
  --version  show zsh version number, then exit
  -b         end option processing, like --
  -c         take first argument as a command to execute
  -o OPTION  set an option by name (see below)
```

**Reply by wez** [reply]
try: 

args = { "zsh", "-c", "lvim" }

**Reply by cynicaljoy** [reply]
see [previous comment](https://github.com/wez/wezterm/discussions/628?sort=top#discussioncomment-6848980) from @wez I found in another discussion that's pertinent for anybody following along

**Reply by karimlevallois** [reply]
Thanks so much for all your help on this, got it working with. "lvim" alone didn't work.

```
wezterm.on("open-lvim", function(window, _)
	local _, _, lvim_window = mux.spawn_window({
		args = { "zsh", "-c", "/Users/Karim/.local/bin/lvim" },
	})
	local overrides = window:get_config_overrides() or {}
	overrides.font_size = 14
	lvim_window:gui_window():set_config_overrides(overrides)
	lvim_window:gui_window():toggle_fullscreen()
end)
```

I think I can finally move away from Kitty!

**Reply by Demianeen** [reply]
@karimlevallois, sorry for the out-of-scope question, but I am really curious how did you get a beautiful start page like this one in the terminal?
![CleanShot 2024-03-10 at 21 02 28@2x](https://github.com/wez/wezterm/assets/51330172/f6a98750-0855-429c-86ce-5497e4136cbd)

**Reply by karimlevallois** [reply]
That’s fastfetch with a little custom tweaking and catppuccin theme. I can share my config tomorrow when I am next at my laptop :)

**Reply by karimlevallois** [reply]
@Demianeen Sorry for the delay, here it is...

```
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
    "logo": {
        "source": "macos2_small",
        "padding": {
            "left": 1,
            "right": 3
        }
    },
    "display": {
        "color": "cyan",
        "separator": " >  "
    },
    "modules": [
        {
            "key": "Machine    ",
            "type": "host"
        },
        {
            "key": "OS         ",
            "type": "os"
        },
        {
            "key": "Packages   ",
            "type": "packages"
        },
        {
            "key": "Terminal   ",
            "type": "terminal",
            // "format": "WezTerm"
        },
        {
            "key": "Resolution ",
            "compactType": "original",
            "type": "display"
        },
        {
            "key": "CPU        ",
            "type": "cpu"
        },
        {
            "key": "Memory     ",
            "type": "memory"
        },
        "break",
        {
            "type": "custom",
            "format": "\u001b[30m  \u001b[31m  \u001b[32m  \u001b[33m  \u001b[34m  \u001b[35m  \u001b[36m  \u001b[37m "
        },
    ]
}

```

**Reply by Demianeen** [reply]
Big thanks @karimlevallois !
